### DISCUSSION TITLE
Is it possible to skip the CI pipeline entirely with `npx turbo-ignore`?

### DISCUSSION THREAD

**Discussion by Haberkamp** [vercel/turborepo#7365] [Help] [answered]
I found this article in the [turborepo docs](https://turbo.build/repo/docs/core-concepts/monorepos/skipping-tasks#customizing-behavior-1). I not only want to cache the task itself but skip the package installation and the whole workflow itself if there is no need to run the action. But wasn't able to successfully integrate this with a GitHub Workflow. So my question is: Is this possible with Github Workflows and if yes, how can I do that?

**Comment by thexpand** [comment]
@Haberkamp Did you find a way to do this?

**Comment by anthonyshew** [comment]
The article you've linked describes what you're looking for. You would want to implement logic that skips the rest of the Action for a `1` exit code. Otherwise, you would execute the rest of the Action.

The rest of the implementation is highly specific to what you're trying to accomplish so I'm not sure there is too much more we could add to the documentation. Is there something specific that you're confused by?

**Reply by thexpand** [reply]
I have a GitHub Actions workflow that is being triggered when a PR is opened. There are 5 applications inside the workflow and all of them start building together (different jobs). How can I build only those that have changes in them using `turbo-ignore`?

**Comment by mehulkar** [comment, marked as answer]
You can model this with 2 workflows (or 2 jobs in the same workflow, or 2 steps in the same job) by running `--dry` first, parsing the results and checking for cache misses, and then using that result to do a real run with `--filter` flag. You also _should_ be able to skip the install step if you first install global turbo instead of using the local one.

We don't have a template or guide for this at the moment, but we use this approach internally at Vercel in some cases as well.
